python/ray/serve/_private/proxy.py

```diff
@@ -640,8 +640,13 @@ async def unary_unary(
                 else:
                     response = message
 
-            context.set_code(status.code)
-            context.set_details(status.message)
+            # Only the latest code and details will take effect. If the user already
+            # set them in the context, skip setting them with Serve's default values.
+            if not context.code():
+                context.set_code(status.code)
+            if not context.details():
+                context.set_details(status.message)
```

Comment:
Could you make the conditions more explicit? E.g. instead of `if not context.details()`, maybe `if context.details() is not None`?

E.g. if the user sets `details=""`, then `if not context.details()` evaluates to `True`, and the details will be overwritten by a default value. I'm not sure if that's expected behavior or not.

Reply:
Oh that's on purpose. If nothing is set, code will be 0 (ok status code in gRPC) and details will be empty string regardless. It also makes no sense for users to set to the default values as well since that doesn't provide any additional information. Will leave those as is :)

Reply:
Synced offline. We'll add a comment here.